Model/Push.php

```diff
@@ -1557,6 +1561,38 @@ private function receivePushCheckPayLink($response, $validSignature)
         return false;
     }
 
+    private function pushCheckPayPerEmailCancel($response, $validSignature, $payment)
+    {
+        $failedStatuses = [
+            'BUCKAROO_MAGENTO2_STATUSCODE_TECHNICAL_ERROR',
+            'BUCKAROO_MAGENTO2_STATUSCODE_VALIDATION_FAILURE',
+            'BUCKAROO_MAGENTO2_STATUSCODE_CANCELLED_BY_MERCHANT',
+            'BUCKAROO_MAGENTO2_STATUSCODE_CANCELLED_BY_USER',
+            'BUCKAROO_MAGENTO2_STATUSCODE_FAILED',
+            'BUCKAROO_MAGENTO2_STATUSCODE_REJECTED'
+        ];
+        if (
+            (
+                isset($this->originalPostData['ADD_fromPayPerEmail'])
```

Comment:
how is about to use $this->postData with lowercased values to prevent possible fail on different push types ?